opm/parser/eclipse/Utility/String.hpp

```diff
@@ -50,5 +52,57 @@ std::string trim_copy(const T& s)
 {
     return ltrim_copy( rtrim_copy(s) );
 }
+
+
+template<typename T>
+void replaceAll(T& data, const T& toSearch, const T& replace)
+{
+    // Get the first occurrence
+    size_t pos = data.find(toSearch);
+
+    // Repeat till end is reached
+    while (pos != std::string::npos)
+    {
+        // Replace this occurrence of Sub String
+        data.replace(pos, toSearch.size(), replace);
+        // Get the next occurrence from the current position
+        pos = data.find(toSearch, pos + replace.size());
+    }
+}
+
+
+inline std::vector<std::string> split_string(const std::string& input,
+                                             char delimiter)
+{
+    std::vector<std::string> result;
+    std::string token;
+    std::istringstream tokenStream(input);
+    while (std::getline(tokenStream, token, delimiter))
+        result.push_back(token);
+
+    return result;
 }
+
+
+inline std::vector<std::string> split_string(const std::string& input,
+                                             const std::string& delimiters)
+{
+    std::vector<std::string> result;
+    std::string::size_type start = 0;
+    while (start < input.size()) {
+        auto end = input.find_first_of(delimiters, start);
+        if (end == std::string::npos) {
+            result.push_back(input.substr(start));
+            start = input.size();
+        } else {
+          result.push_back(input.substr(start, end-start));
+          start = end + 1;
+        }
+    }
+
+    return result;
+}
```

Comment:
You can get more compact code for this if you use a "regex token iterator":
```C++
inline std::vector<std::string>
split_string(const std::string& s, const std::string& delimiters)
{
    if (input.empty()) {
        // Single element vector whose only element is the empty string.
        return { "" };
    }

    if (delimiter.empty()) {
        return { input };
    }

    const auto sep = std::regex(delimiter);

    using TI = std::sregex_token_iterator;

    return { TI(input.begin(), input.end(), sep, -1), TI{} };
}
```

Reply:
The `std::sregex_token_iterator` was very interesting functionality; but I don't think the `sep = std::regex(delimiters)` captures the "split on any of the characters in the `delimiters` string" behavior.

Reply:
as joakim points out that would mean caller had to use a regex. it's not trivial to convert the input chars to regex since they are typically exactly the kind of things that needs escaping.